src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp

```diff
@@ -235,7 +235,7 @@ void MergeTreeBackgroundExecutor<Queue>::routine(TaskRuntimeDataPtr item)
         active.erase(std::remove(active.begin(), active.end(), item_), active.end());
     };
 
-    auto release_task = [] (TaskRuntimeDataPtr && item_) TSA_REQUIRES(mutex)
+    auto release_task = [this] (TaskRuntimeDataPtr && item_) TSA_REQUIRES(mutex)
```

Comment:
Maybe it is better to wrap all methods of the underlying task? For example `cancel`/`wait`/`executeStep`/`reset` and update profile events there?